model-optimizer/unit_tests/mo/back/moc_preprocessing_test_actual.py

```diff
@@ -0,0 +1,617 @@
+    input2.get_output_tensor(0).set_names({'input2', 'input2a'})
+    relu2 = ops.relu(input2)
+    res2 = ops.result(relu2, "res2")
+    res2.get_output_tensor(0).set_names({'res2', 'res2a'})
+    function = Function(results=[res1, res2], parameters=[input1, input2], name="TestFunction")
+    return function
+
+
+def create_function1(shape1=[2, 2]):
+    input1 = ops.parameter(shape1, dtype=np.float32, name="input1")
+    input1.get_output_tensor(0).set_names({'input1', 'input1a'})
+    relu1 = ops.relu(input1)
+    res1 = ops.result(relu1, "res1")
+    res1.get_output_tensor(0).set_names({'res1', 'res1a'})
+    function = Function(results=[res1], parameters=[input1], name="TestFunction")
+    return function
+
+
+def process_function(ov_function: Function, argv: Namespace):
+    apply_preprocessing(ov_function=ov_function, argv=argv)
+
+
+class TestPreprocessingMOC(unittest.TestCase):
+    def setUp(self):
+        pass
+
+    def check_scale_constant(self, node, expected, shape=None):
+        const_node = node.input(1).get_source_output().get_node()
+        self.assertEqual(const_node.get_type_name(), 'Constant')
+        if node.get_type_name() == 'Divide':
+            self.assertTrue(np.allclose(const_node.get_vector(), expected))
+        else:
+            self.assertTrue(np.allclose(const_node.get_vector(), 1. / expected))
+        if shape:
+            assert const_node.shape == PartialShape(shape)
+
+    def check_mean_constant(self, node, expected, shape=None):
+        const_node = node.input(1).get_source_output().get_node()
+        self.assertEqual(const_node.get_type_name(), 'Constant')
+        if node.get_type_name() == 'Subtract':
+            self.assertTrue(np.allclose(const_node.get_vector(), expected))
+        else:
+            self.assertTrue(np.allclose(const_node.get_vector(), -expected.toList()))
+        if shape:
+            self.assertEqual(const_node.shape, PartialShape(shape))
+
+    def test_scale_single_value(self):
+        argv = Namespace(mean_scale_values=None, scale=2.0)
+        function = create_function2()
+        process_function(ov_function=function, argv=argv)
+
+        for param in function.get_parameters():
+            op_node = list(param.output(0).get_target_inputs())[0].get_node()
+            self.assertTrue(op_node.get_type_name() == 'Divide' or op_node.get_type_name() == 'Multiply')
+            self.check_scale_constant(op_node, [2.0])
+
+    def test_scale_single_value_fp64(self):
+        argv = Namespace(mean_scale_values=None, scale=2.0)
+        function = create_function2(dtype1=np.float64)
+        process_function(ov_function=function, argv=argv)
+
+        for ov_input in function.inputs:
+            op_node = list(ov_input.get_target_inputs())[0].get_node()
+            self.assertTrue(op_node.get_type_name() == 'Divide' or op_node.get_type_name() == 'Multiply')
+            self.check_scale_constant(op_node, [2.0])
+
```

Comment:
please add a test case when it specifies both scale_values and scale parameters. I expect to have exception because simultaneous usage is prohibited

Reply:
This case is covered by "main.py" during command line arguments parsing. Current "apply_preprocessing" function doesn't deal with parsing of command line arguments and expects that 'scale' option is already not conflicting with 'scale_values'. Otherwise it will be impossible to hit into this check in real mo.py usage scenario